core/src/main/java/com/amplitude/core/utilities/Diagnostics.kt

```diff
@@ -37,11 +41,13 @@ class Diagnostics() {
             diagnostics["malformed_events"] = malformedEvents!!
         }
         if (errorLogs != null && errorLogs!!.isNotEmpty()) {
-            diagnostics["error_logs"] = errorLogs!!
+            // pick the first MAX_ERROR_LOGS from the set and convert to list
+            val errorLogsToTake = errorLogs!!.take(MAX_ERROR_LOGS)
```

Comment:
Should we limit the error logs when we add error logs to the set? If the count is greater than max count, there is no need to accumulate more in memory right?

Reply:
Sure, I can change it to that way as well. I considered this first, then still wants to take error log if unique error logs are more than limit. If we were to worry about the set uniqueness is not enough, I could just limit how many we allowed in our log.

Reply:
Yea I am worried that an error log having a UUID in it will make the uniqueness not sufficient. And in those cases we'll run into a lot of data being saved in memory.

Reply:
Sounds good, updated.